### PR TITLE
Yyhaier

### PR DESCRIPTION
**Pull request by yuhai-yang**
subgraph demo

### CODE REVIEW

**Comment by liberhe** [comment]
> subgraph demo

Great，please read the link https://github.com/rebase-network/Dapp-Learning/blob/main/docs/dao.md
add WeChat to join our developer group

**Review by maroonstar — APPROVED** [review]

**Review by Loverush — APPROVED** [review]

**Comment by yanyanho** [comment]
> subgraph demo

It is not recommended to  use bsc testnet .  Ethereum , with layer2 are more welcome.

File: basic/34-subgraph/bscdemo/README.md [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+
+将```  ethereum: 'mainnet:http://host.docker.internal:8545'``` 修改成BSC测试链的RPC地址```ethereum: 'bsc:https://data-seed-prebsc-1-s1.binance.org:8545'```
+
+```
+docker-compose up
+```
+
+🎉安装完成。（它需要打开端口 4001、5001、8000、8001、8020、8080 和 5432，因为它将启动一个 postgre sql、一个 ipfs 本地节点和 Graph 节点本身，出现报错首先排查端口是否被占用。）
+
+![image-20211102165025497](/Users/yyhaier/Documents/%E5%85%AC%E5%8F%B8%E9%A1%B9%E7%9B%AE/%E4%BB%8E%E9%9B%B6%E5%BC%80%E5%A7%8B%E6%90%AD%E5%BB%BAGraph%20Node%E9%80%9A%E8%BF%87%20GraphQL%20%E4%BB%8E%E5%8C%BA%E5%9D%97%E9%93%BE%E6%9F%A5%E8%AF%A2%E5%A4%8D%E6%9D%82%E6%95%B0%E6%8D%AE.assets/image-20211102165025497-5843027.png)
```
Comment (jhfnetboy) [inline basic/34-subgraph/bscdemo/README.md]: Please upload the local image~

File: basic/34-subgraph/bscdemo/README.md [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+   yarn global add @graphprotocol/graph-cli
+   ```
+
+2. 安装依赖
+
+   ```
+   yarn && yarn codegen
+   ```
+
+   ![image-20211104180150595](/Users/yyhaier/Documents/%E5%85%AC%E5%8F%B8%E9%A1%B9%E7%9B%AE/%E4%BB%8E%E9%9B%B6%E5%BC%80%E5%A7%8B%E6%90%AD%E5%BB%BAGraph%20Node%E9%80%9A%E8%BF%87%20GraphQL%20%E4%BB%8E%E5%8C%BA%E5%9D%97%E9%93%BE%E6%9F%A5%E8%AF%A2%E5%A4%8D%E6%9D%82%E6%95%B0%E6%8D%AE.assets/image-20211104180150595.png)
```
Comment (jhfnetboy) [inline basic/34-subgraph/bscdemo/README.md]: same problem, or your can use mac app iPic,to upload image to free web space, and paste the url here~

File: basic/34-subgraph/bscdemo/README.md [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+
+3. 分配subgraph 名字
+
+   ```
+   yarn create-local
+   ```
+
+   ![image-20211104180251303](/Users/yyhaier/Documents/%E5%85%AC%E5%8F%B8%E9%A1%B9%E7%9B%AE/%E4%BB%8E%E9%9B%B6%E5%BC%80%E5%A7%8B%E6%90%AD%E5%BB%BAGraph%20Node%E9%80%9A%E8%BF%87%20GraphQL%20%E4%BB%8E%E5%8C%BA%E5%9D%97%E9%93%BE%E6%9F%A5%E8%AF%A2%E5%A4%8D%E6%9D%82%E6%95%B0%E6%8D%AE.assets/image-20211104180251303-6020172.png)
```
Comment (jhfnetboy) [inline basic/34-subgraph/bscdemo/README.md]: The same~

File: basic/34-subgraph/bscdemo/README.md [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+
+4. 本地部署
+
+   ```
+   yarn deploy-local
+   ```
+
+   ![image-20211104180333944](/Users/yyhaier/Documents/%E5%85%AC%E5%8F%B8%E9%A1%B9%E7%9B%AE/%E4%BB%8E%E9%9B%B6%E5%BC%80%E5%A7%8B%E6%90%AD%E5%BB%BAGraph%20Node%E9%80%9A%E8%BF%87%20GraphQL%20%E4%BB%8E%E5%8C%BA%E5%9D%97%E9%93%BE%E6%9F%A5%E8%AF%A2%E5%A4%8D%E6%9D%82%E6%95%B0%E6%8D%AE.assets/image-20211104180333944-6020214.png)
```
Comment (jhfnetboy) [inline basic/34-subgraph/bscdemo/README.md]: The same

File: basic/34-subgraph/bscdemo/README.md [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+
+至此部署完成。
+
+## 四、测试
+
+打开GraphQl图形化界面 “http://127.0.0.1:8000/subgraphs/name/myBSCGraph”
+
+![image-20211104180550974](/Users/yyhaier/Documents/%E5%85%AC%E5%8F%B8%E9%A1%B9%E7%9B%AE/%E4%BB%8E%E9%9B%B6%E5%BC%80%E5%A7%8B%E6%90%AD%E5%BB%BAGraph%20Node%E9%80%9A%E8%BF%87%20GraphQL%20%E4%BB%8E%E5%8C%BA%E5%9D%97%E9%93%BE%E6%9F%A5%E8%AF%A2%E5%A4%8D%E6%9D%82%E6%95%B0%E6%8D%AE.assets/image-20211104180550974-6020352.png)
```
Comment (jhfnetboy) [inline basic/34-subgraph/bscdemo/README.md]: The same

File: basic/34-subgraph/bscdemo/README.md [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+
+随机选一组数据进行验证，例如：第[21004](https://testnet.bscscan.com/block/21004)区块
+
+![image-20211104180700191](/Users/yyhaier/Documents/%E5%85%AC%E5%8F%B8%E9%A1%B9%E7%9B%AE/%E4%BB%8E%E9%9B%B6%E5%BC%80%E5%A7%8B%E6%90%AD%E5%BB%BAGraph%20Node%E9%80%9A%E8%BF%87%20GraphQL%20%E4%BB%8E%E5%8C%BA%E5%9D%97%E9%93%BE%E6%9F%A5%E8%AF%A2%E5%A4%8D%E6%9D%82%E6%95%B0%E6%8D%AE.assets/image-20211104180700191-6020421.png)
```
Comment (jhfnetboy) [inline basic/34-subgraph/bscdemo/README.md]: The same

**Pull request merged**